### PR TITLE
Feat: Function to assert whether manifest file contains PackageReference

### PR DESCRIPTION
**Pull request by justshiv**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/dotnet-deps-parser/blob/master/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adds function to assert whether a project file contains any `PackageReference` 

#### Any background context you want to provide? 
(Quoted from the ticket)
In order to understand correctly, which file should be used to get dependency tree, we need to know if user have migrated to the new version of dependency management. We can recognise it by the presence of at least one `<PackageReference>` node in `*.[cfsvb]proj`.

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/BST-315

### CODE REVIEW

File: test/fixtures/dotnet-no-packagereference/project.csproj [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (justshiv) [inline test/fixtures/dotnet-no-packagereference/project.csproj]: Manifest files without `PackageReference` to show false behaviour

File: lib/index.ts [hunk 2]
```diff
@@ -120,3 +121,20 @@ async function extractTargetFrameworksFromProjectConfig(
...
+  }
+
+  const manifestFileFullPath = path.resolve(root, manifestFilePath);
+
+  if (!fs.existsSync(manifestFileFullPath)) {
+    throw new Error('No project file found at ' +
+      `location: ${manifestFileFullPath}`);
+  }
+
+  const manifestFileContents = fs.readFileSync(manifestFileFullPath, 'utf-8');
```
Comment (justshiv) [inline lib/index.ts]: Reading in file as I assume we just need text representation of this. We're not needing to parse the XML as I understand, as we just want to know true/false and not the exact list of `PackageReference` items.

File: lib/index.ts [hunk 3]
```diff
@@ -120,3 +121,13 @@ async function extractTargetFrameworksFromProjectConfig(
     throw new Error(`Extracting target framework failed with error ${err.message}`);
   }
 }
+
+async function containsPackageReference(manifestFileContents: string) {
+
+  const manifestFile: any = await parseManifestFile(manifestFileContents);
```
Comment (yuliabaron) [inline lib/index.ts]: my only comment here is performance. we parse packages.config to decide which manifest file to pick, and then we'll parse it again. Not sure how to improve it though

Comment (justshiv) [reply lib/index.ts]: This was my main concern as well. I originally parsed it just as text to find the reference but Mila pointed out that xml can have comments that could include the PackageReference as a word.

Comment (lili2311) [reply lib/index.ts]: From some of my testing the parsing was pretty fast alone, let's revisit if it shows to be a problem

**Pull request merged**